sdk/python/feast/repo_config.py

```diff
@@ -235,11 +207,33 @@ def get_online_config_from_type(online_store_type: str):
         # This can only be one type of error, when class_name attribute does not exist in the module
         # So we don't have to include the original exception here
         raise errors.FeastClassImportError(
-            module_name, config_class_name, class_type="OnlineStoreConfig"
+            module_name, config_class_name, class_type=f"{store_type}Config"
         ) from None
     return online_store_config_class
 
 
+def get_online_config_from_type(online_store_type: str):
+    if online_store_type in ONLINE_STORE_CLASS_FOR_TYPE:
+        online_store_type = ONLINE_STORE_CLASS_FOR_TYPE[online_store_type]
+    else:
+        assert online_store_type.endswith("OnlineStore")
+    module_name, online_store_class_type = online_store_type.rsplit(".", 1)
+    config_class_name = f"{online_store_class_type}Config"
+
+    return get_config_class_from_type(module_name, config_class_name, "OnlineStore")
+
+
+def get_offline_config_from_type(offline_store_type: str):
+    if offline_store_type in OFFLINE_STORE_CLASS_FOR_TYPE:
+        offline_store_type = OFFLINE_STORE_CLASS_FOR_TYPE[offline_store_type]
+    else:
+        assert offline_store_type.endswith("OfflineStore")
+    module_name, offline_store_class_type = offline_store_type.rsplit(".", 1)
+    config_class_name = f"{offline_store_class_type}Config"
+
+    return get_config_class_from_type(module_name, config_class_name, "OfflineStore")
```

Comment:
These 2 can also be deduplicated. For example, you could construct
```
STORE_CLASS_FOR_TYPE = {
    "OnlineStore": ONLINE_STORE_CLASS_FOR_TYPE,
    "OfflineStore": OFFLINE_STORE_CLASS_FOR_TYPE,
}

def get_config_class_from_type(store_type: str, store_class_type: str):
    if store_class_type in STORE_CLASS_FOR_TYPE[store_type]:
        store_class_type = STORE_CLASS_FOR_TYPE[store_type][store_class_type]
    else:
        assert store_class_type.endswith(store_type)
    module_name, store_class_type = store_class_type.rsplit(".", 1)
    config_class_name = f"{store_class_type}Config"

    # merge existing get_config_class_from_type function (line 188) here instead of splitting
    return get_config_class_from_type(module_name, config_class_name, store_type)
```

Ideally, we shouldn't have Providers handled in a different way, but it's a little bit different (e.g. no Config classes there), so I'll leave it up to you if you think we can deduplicate that in a reasonable way.

Reply:
I pulled in the provider to use a shared code path for loading.

I'm tempted to not introduce `STORE_CLASS_FOR_TYPE` cause the additional indirection/method doesn't seem that bad, and seems more legible. WDYT?

Reply:
Take a look at the latest version, it's more cleaned up